qiskit/backends/ibmq/ibmqjob.py

```diff
@@ -311,31 +386,38 @@ def _wait_for_job(self, timeout=60, wait=5):
                                   (pprint.pformat(api_result)))
             if (api_result['status'] == 'ERROR_CREATING_JOB' or
                     api_result['status'] == 'ERROR_RUNNING_JOB'):
-                job_result = {'job_id': job_id, 'status': 'ERROR',
+                job_result = {'id': id, 'status': 'ERROR',
                               'result': api_result['status']}
-                return Result(job_result, qobj)
+                return Result(job_result)
 
         if self.cancelled:
```

Comment:
This will send another request to the server, I would just check against api_result['status']. There's no way a job can be canceled by the user while is calling `result`, as this is a blocking call.